prov/efa/src/rxr/rxr_init.c

```diff
@@ -312,11 +312,17 @@ void rxr_set_shm_hints(struct fi_info *shm_hints)
 	shm_hints->fabric_attr->prov_name = strdup("shm");
 	shm_hints->ep_attr->type = FI_EP_RDM;
 
-#if HAVE_LIBCUDA
-	shm_hints->caps |= FI_HMEM;
-	shm_hints->domain_attr->mr_mode |= FI_MR_HMEM;
-#endif
-
+	/*
+	 * We validate whether FI_HMEM is supported before this function is
+	 * called, so it's safe to check for this via the app hints directly.
+	 * We should combine this and the earlier FI_HMEM validation when we
+	 * clean up the getinfo path. That's not possible at the moment as we
+	 * only have one SHM info for the entire provider which isn't right.
+	 */
+	if (app_hints && app_hints->caps & FI_HMEM) {
```

Comment:
I would prefer add parenthesis around `app_hints->caps & FI_HMEM` in this case.

Reply:
Yup, good call. Fixed.